### DISCUSSION TITLE
Plans on the roadmap to extend this extension to use Postgres's Foreign Data Wrapper?

### DISCUSSION THREAD

**Discussion by ramarnat** [duckdb/pg_duckdb#167] [General]
Is it on the roadmap to do something with foreign data wrapper functionality in Postgres? Here is a [repo](https://github.com/alitrack/duckdb_fdw) that has that functionality, maybe something to integrate into this official extension? 

ParadeDB's [pg_analytics](https://github.com/paradedb/pg_analytics) also has implemented it as a foreign wrapper, and is a more sophisticated implementation, albeit with more restrictive licensing.

**Comment by zhaonaiy** [comment]
+1, more specific, to use this engine to read EXISTing FDW of PG.

**Reply by JelteF** [reply]
Now that #477 is merged it should be possible to read data from existing FDWs

**Comment by JelteF** [comment]
For now we're focusing on the "query-stealing" approach, i.e. let DuckDB take over planning and execution completely. We might integrate FDWs in the future, but that's not planned for the near-term.
